app/src/main/kotlin/com/hippo/ehviewer/spider/SpiderDen.kt

```diff
@@ -218,6 +225,41 @@ class SpiderDen(mGalleryInfo: GalleryInfo) {
             override fun close() {}
         }
     }
+
+    suspend fun exportAsCbz(file: UniFile) = resourceScope {
+        val pages = info.pages
+        val comicInfo = closeable {
+            val f = downloadDir!! / COMIC_INFO_FILE
+            if (!f.exists()) writeComicInfo()
+            f.openFileDescriptor("r")
+        }
+        val (fdBatch, names) = (0 until pages).parMap { idx ->
+            val ext = requireNotNull(getExtension(idx))
+            val f = autoCloseable { requireNotNull(getImageSource(idx)) }
```

Comment:
This will crash the app, should use cancellation instead.
Also calling `findImageFile` twice in `getExtension` and `getImageSource` is not ideal.